pkg/eventmanager/eventmanager.go

```diff
@@ -12,7 +11,10 @@ import (
 )
 
 const (
-	REFRESH_INTERVAL = 5 * time.Minute
+	UPGRADE_PRECHECK_FAILED_DESC = "Upgrade failed as the cluster did not pass its pre-upgrade verification checks."
+	UPGRADE_PREHEALTHCHECK_FAILED_DESC = "Upgrade failed on the Pre-Health Check step. Health alerts are firing in the cluster which could impact the upgrade's operation, so the upgrade did not proceed."
+	UPGRADE_EXTDEPCHECK_FAILED_DESC = "Upgrade failed on the External Dependency Availability Check step. A dependency of the upgrade was deemed un-available, so the upgrade did not proceed."
+	UPGRADE_SCALE_FAILED_DESC = "Upgrade failed on the Scale-Up Worker Node step. A temporary additional worker node was unable to be created to temporarily house workloads, so the upgrade did not proceed."
```

Comment:
So far, the message description mentions of "something is wrong". Is it possible to add "Refer SOP to troubleshoot/fix"? It might add bit of work but will greatly help in anyone to look at the right place after an error occurs.

Reply:
A couple of things to unpack here which will hopefully explain better. This isn't intended as SRE messaging - this description is intended to communicate additional failure context to OCM/ClusterService, which will use it to provide more informative service log messaging to the cluster owner. (whether the text is used verbatim, or merely as an indication for OCM to apply its own templating, is TBD)

All of these failure states already raise alerts which arrive in PD with appropriate SOP links attached.. so the SRE flow for handling these states should already be clearly communicated.

Reply:
Understood, but the customer needs to know what action to take (if any). I recommend we merge this update as-is and file a task for adding a KB article to guide customer on follow-up actions (if any). This KB should have all the known failure cases documented on what to do. Once the KB lands we update this with a link.

Reply:
I've updated the messaging to provide clarification to the customer about what each event means and what actions they need to take, to hopefully alleviate some of this. Definitely agree that adding a doco article link will be the good end solution here. I've raised a ticket with the doc team around this.